client/src/components/organisms/RequestGroupForm.tsx

```diff
@@ -285,15 +279,57 @@ const RequestGroupForm: FunctionComponent<Props> = (props: Props) => {
         const tempNameError = updateNameError(name);
         const tempDescriptionError = updateDescriptionError(description);
         const tempImageError = updateImageError(image);
-        const tempRequestTypesError = updateRequestTypeNamesError(requestTypeNames);
+        const tempRequestTypesError = updateRequestTypesMapError(requestTypesMap);
 
         if (!tempNameError && !tempDescriptionError && !tempImageError && !tempRequestTypesError) {
             if (props.operation === "create") {
-                createRequestGroup({ variables: { name, description, image, requestTypeNames } });
+                let requestGroupId: string | null = null;
+                const newRequestTypeNames = Array.from(requestTypesMap)
+                    .filter(([_requestTypeName, requestTypeData]) => requestTypeData.deleted === false)
+                    .map(([requestTypeName, _requestTypeData]) => requestTypeName);
+                createRequestGroup({ variables: { name, description, image } })
+                    .then((response) => {
+                        requestGroupId = response.data.createRequestGroup._id;
+                        const createRequestTypePromises = newRequestTypeNames.map((requestTypeName) =>
+                            createRequestType({ variables: { name: requestTypeName, requestGroupId } })
+                        );
+                        return Promise.all(createRequestTypePromises);
+                    })
+                    .then(() => {
+                        props.onSubmitComplete();
+                    });
             } else {
-                updateRequestGroup({
-                    variables: { id: props.requestGroupId, name, description, image, requestTypeNames }
-                });
+                const newRequestTypeNames = Array.from(requestTypesMap)
+                    .filter(
+                        ([_requestTypeName, requestTypeData]) =>
+                            requestTypeData.id == null && requestTypeData.deleted === false
+                    )
+                    .map(([requestTypeName, _requestTypeData]) => requestTypeName);
+                const deletedRequestTypeIds = Array.from(requestTypesMap.values())
```

Comment:
need to review deleting requestTypes in the future - maybe can do "concurrently" with creating requestTypes